python/Gaffer/Application.py

```diff
@@ -86,6 +94,10 @@ def root( self ) :
 	## Called to run the application and return a status value.
 	def run( self ) :
 
+		if self.parameters()["help"].getTypedValue() :
+			self.__formatHelp()
+			return
```

Comment:
I think this should be `return 0` since `run()` is meant to return a status value.